hi-ml-azure/testazure/testazure/test_himl.py

```diff
@@ -119,6 +120,103 @@ def test_write_run_recovery_file(mock_run: mock.MagicMock) -> None:
     assert expected_run_recovery_id != recovery_file_text
 
 
+@pytest.fixture(scope="module")
+def dummy_max_num_nodes_available() -> int:
+    return randint(2, 10)
+
+
+@pytest.fixture(scope="module")
+def dummy_compute_cluster_name() -> str:
+    return 'dummy_cluster'
+
+
+@pytest.fixture(scope="module")
+def mock_scale_settings(dummy_max_num_nodes_available: int) -> MagicMock:
+    return MagicMock(maximum_node_count=dummy_max_num_nodes_available)
+
+
+@pytest.fixture(scope="module")
+def mock_compute_cluster(mock_scale_settings: MagicMock) -> MagicMock:
+    return MagicMock(scale_settings=mock_scale_settings)
+
+
+@pytest.fixture(scope="module")
+def mock_workspace(mock_compute_cluster: MagicMock, dummy_compute_cluster_name: str) -> MagicMock:
+    return MagicMock(compute_targets={dummy_compute_cluster_name: mock_compute_cluster})
+
+
+@pytest.mark.fast
+def test_validate_num_nodes(dummy_max_num_nodes_available: int, dummy_compute_cluster_name: str,
+                            mock_workspace: MagicMock) -> None:
+    # If number of requested nodes <= max available nodes, nothing should happen
+    num_nodes_requested = dummy_max_num_nodes_available // 2
+    existing_compute_clusters = mock_workspace.compute_targets
+    print(f"Existing compute cluster scale settings: {existing_compute_clusters['dummy_cluster'].scale_settings}")
```

Comment:
Do we need print statements in tests?

Reply:
I see no issues with those

Reply:
Happy to remove, probably not a very useful statement anyway